src/js/searchModal/searchYoutube.js

```diff
@@ -0,0 +1,150 @@
+import api from 'apis/api';
+import {
+  renderYoutubeItems,
+  renderNotFoundMessage,
+  renderSkeleton,
+  hideSkeleton,
+  refreshItems,
+  renderChips,
+  renderSnackBar,
+  renderDependsOnTabState,
+} from 'utils/render';
+import { iterateWithIsSavedState } from 'utils/iterate';
+import fetchVideoId from 'utils/fetchVideoId';
+
+const $modalVideoWrapper = document.querySelector('.modal .video-wrapper');
+const $searchInput = document.querySelector('.modal form .w-100');
+const $searchForm = document.querySelector('.modal form');
+const $modalInner = document.querySelector('.modal-inner');
+const $savedCount = document.querySelector('.saved-count');
+const $mainVideoWrapper = document.querySelector('main .video-wrapper');
+const $div = document.createElement('div');
+$div.setAttribute('id', 'modalFetchMore');
+
+const searchInfo = { nextPage: '', value: '', items: [] };
+const initialSavedItems = JSON.parse(localStorage.getItem('savedItems'));
+let savedItems = [];
+let savedCount = initialSavedItems ? initialSavedItems.length : 0;
+let chips = JSON.parse(localStorage.getItem('lastestSearches')) || [];
+
+$savedCount.innerText = initialSavedItems ? initialSavedItems.length : '0';
+
+const checkIfDuplicates = (value, cond) => {
+  if (chips.includes(value)) {
+    chips.splice(chips.indexOf(value), 1);
+    chips = [...chips, value];
+    localStorage.setItem('lastestSearches', JSON.stringify(chips));
+  } else {
+    cond && chips.shift();
+    chips = [...chips, value];
+    localStorage.setItem('lastestSearches', JSON.stringify(chips));
+  }
+};
```

Comment:
함수명이랑 함수 역할이랑 잘 매치가 안돼요 😭 😭 
그리고, splice말고 filter를 사용해보시는건 어떠세요 ? 괜히 메서드가 2개 실행되는 것 같아서요ㅎㅎㅎ
그리고 splice는 chips 자체를 변경시키는 문제도 있구용ㅎㅎ
아니면, 중복을 피하시고 싶으시면 자바스크립트의 Set 자료구조도 사용해보시면 좋으실것 같아요~!
```suggestion
const checkIfDuplicates = (value, cond) => {
  if (chips.includes(value)) {
    chips = chips.filter((v)=>v!==value)).push(value); 
    localStorage.setItem('lastestSearches', JSON.stringify(chips));
  } else {
    cond && chips.shift();
    chips = [...chips, value];
    localStorage.setItem('lastestSearches', JSON.stringify(chips));
  }
};
```

Reply:
오오... 불변성을 지키는 것을 항상 생각은 하고있었는데 splice를 쓰면 원본 배열도 바뀌겠네요! 
Set을 한 번도 활용해본적이 없는데 Set을 한 번 사용해봐야겠네요!!